src/library/scala/collection/mutable/AnyRefMap.scala

```diff
@@ -367,6 +367,20 @@ class AnyRefMap[K <: AnyRef, V] private[collection] (defaultEntry: K => V, initi
     }
   }
 
+  override def foreachKeyValue[U](f: (K,V) => U): Unit = {
```

Comment:
This implementation is not recursive. Unlike most others it may be possible to implement `foreach` in terms of `foreachKeyValue` (or the other way around?) and have the inliner optimize it.

Reply:
The same might work for `mutable.LongMap` and `mutable.OpenHashMap`